From: Sascha Hauer <s.hauer@pengutronix.de>
Date: Thu, 19 Oct 2017 12:13:48 +0200
Subject: [PATCH] graphic_utils: Fix freeing uninitialzed pointer in fb_open()

In fb_open() when fb_create_screen() fails then sc contains an
error pointer which is then freed in the failure path. Fix this.

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 lib/gui/graphic_utils.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/lib/gui/graphic_utils.c b/lib/gui/graphic_utils.c
index d3f5cce4b739..1e3e46b666af 100644
--- a/lib/gui/graphic_utils.c
+++ b/lib/gui/graphic_utils.c
@@ -286,7 +286,7 @@ struct screen *fb_open(const char * fbdev)
 	sc = fb_create_screen(info);
 	if (IS_ERR(sc)) {
 		ret = PTR_ERR(sc);
-		goto failed_create;
+		goto failed_screeninfo;
 	}
 
 	sc->fd = fd;
@@ -294,8 +294,6 @@ struct screen *fb_open(const char * fbdev)
 
 	return sc;
 
-failed_create:
-	free(sc);
 failed_screeninfo:
 	close(fd);
 
